src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsDiagnosticItemSourceProvider.cs

```diff
@@ -7,162 +7,162 @@
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.Internal.VisualStudio.PlatformUI;
 using Microsoft.VisualStudio.ProjectSystem;
 using Microsoft.VisualStudio.ProjectSystem.Properties;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
+
+using OrderAttribute = Microsoft.VisualStudio.Utilities.OrderAttribute;
+using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+[Export(typeof(IAttachedCollectionSourceProvider))]
+[Name(nameof(CpsDiagnosticItemSourceProvider))]
+[Order]
+[AppliesToProject($"({ProjectCapabilities.CSharp} | {ProjectCapabilities.VB}) & {ProjectCapabilities.Cps}")]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CpsDiagnosticItemSourceProvider(
+    IThreadingContext threadingContext,
+    [Import(typeof(AnalyzersCommandHandler))] IAnalyzersCommandHandler commandHandler,
+    IDiagnosticAnalyzerService diagnosticAnalyzerService,
+    VisualStudioWorkspace workspace,
+    IAsynchronousOperationListenerProvider listenerProvider)
+    : AttachedCollectionSourceProvider<IVsHierarchyItem>
 {
-    using OrderAttribute = Microsoft.VisualStudio.Utilities.OrderAttribute;
-    using Workspace = Microsoft.CodeAnalysis.Workspace;
-
-    [Export(typeof(IAttachedCollectionSourceProvider))]
-    [Name(nameof(CpsDiagnosticItemSourceProvider))]
-    [Order]
-    [AppliesToProject($"({ProjectCapabilities.CSharp} | {ProjectCapabilities.VB}) & {ProjectCapabilities.Cps}")]
-    internal sealed class CpsDiagnosticItemSourceProvider : AttachedCollectionSourceProvider<IVsHierarchyItem>
-    {
-        private readonly IAnalyzersCommandHandler _commandHandler;
-        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
-        private readonly Workspace _workspace;
-
-        private IHierarchyItemToProjectIdMap? _projectMap;
-
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public CpsDiagnosticItemSourceProvider(
-            [Import(typeof(AnalyzersCommandHandler))] IAnalyzersCommandHandler commandHandler,
-            IDiagnosticAnalyzerService diagnosticAnalyzerService,
-            VisualStudioWorkspace workspace)
-        {
-            _commandHandler = commandHandler;
-            _diagnosticAnalyzerService = diagnosticAnalyzerService;
-            _workspace = workspace;
-        }
+    private readonly IThreadingContext _threadingContext = threadingContext;
```

Comment:
>_threadingContext [](http://example.com/codeflow?start=39&length=17)

I assume usage of _threadingContext and _listenerProvider is in the associated PR?

Reply:
yes.